spec/elastic_apm_spec.rb

```diff
@@ -152,6 +152,97 @@
     after { ElasticAPM.stop }
   end
 
+  context 'async spans', :intercept do
+    context 'transaction parent' do
+      it 'allows async spans' do
+        with_agent do
+          transaction = ElasticAPM.start_transaction
+          span1 = Thread.new do
+            ElasticAPM.start_span('job 1', parent: transaction)
```

Comment:
Do we want to `end_span` or `with_span` these so they close, just for good measure?

Reply:
done!